__tests__/workers/postUpdated.ts

```diff
@@ -593,3 +611,142 @@ it('should not update already approved post', async () => {
   expect(submission.id).toEqual(uuid);
   expect(submission.status).toEqual(SubmissionStatus.Accepted);
 });
+
+describe('on post create', () => {
+  beforeEach(async () => {
+    await saveFixtures(con, Source, [
+      {
+        id: COMMUNITY_PICKS_SOURCE,
+        name: 'Community recommendations',
+        image: 'sample.image.com',
+      },
+    ]);
+
+    await createDefaultUser();
+  });
+
+  describe('when data includes questions', () => {
+    it('creates a question record each one', async () => {
+      const uuid = randomUUID();
+      await createDefaultSubmission(uuid);
+
+      // pre-check
+      const questionsBefore = await con.getRepository(PostQuestion).find();
+      expect(questionsBefore.length).toEqual(0);
+
+      await expectSuccessfulBackground(worker, {
+        id: 'f99a445f-e2fb-48e8-959c-e02a17f5e816',
+        title: 'With questions',
+        url: `https://post.com/${uuid}`,
+        source_id: COMMUNITY_PICKS_SOURCE,
+        submission_id: uuid,
+        extra: {
+          questions: DEFAULT_QUESTIONS,
+        },
+      });
+
+      const post = await con
+        .getRepository(Post)
+        .findOneBy({ title: 'With questions' });
+      const questionsAfter = await con
+        .getRepository(PostQuestion)
+        .findBy({ postId: post.id });
+      expect(questionsAfter.length).toEqual(3);
+      expect(questionsAfter[0].question).toEqual(
+        'What is the one thing you need to truly understand neural networks?',
+      );
+      expect(questionsAfter[1].question).toEqual(
+        'How do neural networks learn?',
+      );
+      expect(questionsAfter[2].question).toEqual(
+        'What is the process of training a neural network?',
+      );
+    });
+  });
+
+  describe('when data does not include questions', () => {
```

Comment:
This one would already be covered in general as the other test don't include questions.

Reply:
agreed, but since the other tests don't explicitly test for this, I'd like to still keep this one as a sanity check -- in case the other ones get updated in the future to include questions for some reason